### ISSUE TITLE
Relion attempts to download fftw, during configure/compilation step, when there is no network connection available

### ISSUE THREAD

**Issue by shevegen** [3dem/relion#293]
Hello Björn and others,

Not sure if this is worth reporting. However had, I'll still report it but feel free to just close it.

Today I was compiling first openmpi, and then relion.

Relion did not find fftw, I think, and attempted to download it.

With fftw I assume this is meant: http://www.fftw.org/fftw-3.3.6-pl2.tar.gz

I had no working internet connection so the download can never finish successfully. But I have lots of source code archive files in my regular backup available so even my offline computers should be able to compile stuff; e. g. the above package is available locally. In this context, I would have preferred relion to indicate that fftw is missing and stopped. I'd compile fftw next on my own, anyway.

On the other hand, I also understand why there is code that attempts to automatically download; that way, users can proceed to compile relion from source too, right after fftw is available. That part is fine IMO. It's just a strange situation when the above can never work due to missing internet connection (was crappy WLAN/WIFI... would have worked with an ethernet cable but I did not have one at the place where I was with my laptop).

So here are some suggestions:

(1) A more prominent indication of fftw on the commandline either way. Perhaps also some indication as to which configure options fftw may want to have or require; I recall years ago that relion wanted some specific function in fftw or so.
(2) Perhaps an option to, rather than attempt to download, to bail out. There is already code that tries it, e. g. a retry before fftw is downloaded again ... which will not work if the computer is offline though.

Anyway, not sure if the above is worth any time investment but I only wanted to give some feedback.

I used relion in 2014 and it was pretty cool; since then I see that it has much improved, the looks of the GUI are also more sophisticated (I haven't yet tried to do any more work with cryo-electron microscopy since I don't have one at home :P but cryo-electron microscopy recently won the nobel prize so I guess it's a field that is still "evolving forward"). \o/

**Comment by dkimanius** [comment]
Thank you for the feedback.

(1) During configure (cmake run) you get a pretty prominent message saying:
```
-------------------------------------------------
------- WILL USE LOCALLY BUILT FFTW LIBS -------
-------------------------------------------------
```
 I can see that there could be some confusion to whether this implies also downloading the library. I'll add this info into the next major release.

(2) There are cmake options to force building local libraries (e.g. FORCE_OWN_FFTW). However we have no option for the opposite. Rather than giving this option to the user the build process instead will try to recover from any issues by automatically try to download the library for the user. I see no harm in doing so as long as the user is informed beforehand.

**Comment by dkimanius** [comment]
Issue fixed in commit b79058ee4372c8b5c1c72f44d920ee5dbe1e4303
